macros/hardware_functions/mmu.cfg

```diff
@@ -0,0 +1,313 @@
+    {% elif printer.mmu.tool|int != -2 and printer.mmu.tool|int != INITIAL_TOOL %}
+        {action_raise_error("Initial tool ({INITIAL_TOOL}) and MMU tool ({printer.mmu.tool}) are different!!! Check then restart the print. PS: Sorry but it's actually the only way to stop the START_PRINT gcode!!!")}
+    {% elif verbose %}
+        {% if printer.mmu.tool|int == -1 %}
+            RESPOND PREFIX='mmu gate:' MSG="<span class="warning"--text> something get wrong! No tool active! </span>"
+        {% elif printer.mmu.tool|int == -2 %}
+            RESPOND PREFIX='mmu gate:' MSG="<span class="success"--text> Bypass selected and loaded</span>"
+        {% else %}
+            RESPOND PREFIX='mmu tool:' MSG="<span class="success"--text> T{printer.mmu.tool} ready</span>"
+        {% endif %}
+    {% endif %}
+
+
+## Custom macros for MMU in addition to that in HHv2
+
+[gcode_macro MMU_SET_TOOL_TO_GATE]
+description: Macro to remap single Tool to Gate.
+gcode:
+    {% set verbose = printer["gcode_macro _USER_VARIABLES"].verbose %}
+    {% set TOOL = params.TOOL_TO_REMAP|default(-1)|int %}
+    {% set GATE = params.GATE_TO_LINK|default(-1)|int %}
+
+    {% if printer.mmu.enabled %}
+        {% if TOOL == -1 or GATE == -1 %}
+            RESPOND PREFIX='Config info:' MSG="<span class="warning"--text> TOOL and GATE must be provided"
+            MMU_REMAP_TTG
+        {% else %}
+            MMU_REMAP_TTG TOOL={TOOL} GATE={GATE}
+            {% if verbose %}
+                RESPOND PREFIX='Info:' MSG="<span class="success"--text> T{TOOL} is now linked to MMU Gate {GATE}</span>"
+            {% endif %}
+        {% endif %}
+    {% else %}
+        RESPOND PREFIX='Info:' MSG="<span class="error"--text> MMU is disabled!</span>"
+    {% endif %}
+
+[gcode_macro MMU_SET_GATE_SPOOL_ID]
+description: Macro to set Spoolman Spool ID linked to a Gate.
+gcode:
+    {% set verbose = printer["gcode_macro _USER_VARIABLES"].verbose %}
+    {% set spoolman_enabled = printer["gcode_macro _USER_VARIABLES"].spoolman_enabled %}
+    {% set GATE = params.GATE|default(-1)|int %}
+    {% set SPOOL_ID = params.SPOOL_ID|default(-1)|int %}
+
+    {% if spoolman_enabled %}
+        {% if printer.mmu.enabled %}
+            {% if GATE == -1 or SPOOL_ID == -1 %}
+                RESPOND PREFIX='Config info:' MSG="<span class="warning"--text> GATE and SPOOL_ID must be provided"
+                MMU_GATE_MAP
+            {% else %}
+            MMU_GATE_MAP GATE={GATE} SPOOLID={SPOOL_ID}
+                {% if verbose %}
+                    RESPOND PREFIX='Info:' MSG="<span class="success"--text> Gate {GATE} used Spoolman Spool ID: {SPOOL_ID}</span>"
+                {% endif %}
+            {% endif %}
+        {% else %}
+            RESPOND PREFIX='Info:' MSG="<span class="error"--text> MMU is disabled!</span>"
+        {% endif %}
+    {% else %}
+        RESPOND PREFIX='Config info:' MSG="<span class="warning"--text> Spoolman disabled"
+    {% endif %}
```

Comment:
I feel that these two macros are not needed as they do exactly the same thing as the original `MMU_REMAP_TTG` and `MMU_GATE_MAP` macros. The extra check they provide is not really needed since the macros should not exist if no MMU is included.

Do you think it's ok to remove them or is there something I'm missing?

Reply:
Ok I think I understand the need: it's to get them available directly in Fluidd/Mainsail. We can keep these two macros for now.

Reply:
these two macros differ. One is for Tool to Gate (HH only) and the other Gate to Spool (HH with SM). Those both macros do not use same MMU calls, they differ.

Reply:
or you mean it differently and i don't get it

Reply:
ok, after third read i understand :D Sorry

Reply:
i think both ways have their own benefits. 
Question for me: are these macros called only manually?

Reply:
Yes this macros are only called manually and are not used in Klippain.
I made them to simplify for myself (in mainsail... or fluidd) these 2 procedures that I use quite frequently.
If you want we can delete them... and I will add this in my override  because they are not really necessary.

Reply:
The solution I think is to keep them but use the same official name that what is included in HH. This will show them into the WebUI to allow a simple use but will not duplicate the feature to another macro that have the same name.